### PR TITLE
Add Taker Fee Query

### PR DESCRIPTION
**Pull request by mattverse**
Closes: #XXX

## What is the purpose of the change

Adds taker fee query and add it to stargate whitelist. 

Needed for necessary calculations for all contracts that have dependency on calculation. 

## Testing and Verifying

This change is a trivial rework / code cleanup without any test coverage.

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes?
  - [ ] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented? 
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [ ] Code comments?
  - [ ] N/A

### CODE REVIEW

**Comment by mattverse** [comment]
devbot help

File: proto/osmosis/poolmanager/v1beta1/query.proto [hunk 1]
```diff
@@ -119,6 +119,13 @@ service Query {
         "/osmosis/poolmanager/v1beta1/pools/{pool_id}/total_volume";
   }
 
+  // TotalVolumeForPool returns the total volume of the specified pool.
```
Comment (czarcas7ic) [inline proto/osmosis/poolmanager/v1beta1/query.proto:122]:
```suggestion
  // TradingPairTakerFee returns the taker fee for a given set of denoms
```

File: proto/osmosis/poolmanager/v1beta1/query.proto [hunk 2]
```diff
@@ -281,6 +288,19 @@ message TotalVolumeForPoolResponse {
   ];
 }
 
+//=============================== TotalVolumeForPool
```
Comment (czarcas7ic) [inline proto/osmosis/poolmanager/v1beta1/query.proto:291]:
```suggestion
//=============================== TradingPairTakerFee
```

**Review by czarcas7ic — COMMENTED** [review]
Don't we need CLI for this as well as add to the query server? It seems this isn't registered rn

Would also feel good to run this on localosmosis if possible to prove the query works, after adding CLI and registering the message.

**Comment by mattverse** [comment]
I didn't add CLI on purpose(didn't see any need for it), but on the second thought, why not :) 
Added in the most recent commit and comfirmed that it fetches state correctly
```
./osmosisd q poolmanager trading-pair-taker-fee stake uosmo
taker_fee: "0.100000000000000000"
```

**Review by czarcas7ic — APPROVED** [review]

**Pull request merged**